### ISSUE TITLE
Modernize and standardize this project

### ISSUE THREAD

**Issue by rainzee** [ChinaIceF/PyQt-SiliconUI#6]
## Abstract & Motivation

This proposal aims to enhance the standardization and modernization of the component library, focusing on improving the developer experience and overall code quality.

As one of the maintainers of [PyQt-Fluent-Widgets](https://github.com/zhiyiYo/PyQt-Fluent-Widgets), I understand the significant benefits of standardizing and modernizing our codebase. These benefits include improved code writing experiences, enhanced error checking, seamless continuous integration, reduced errors, and performance improvements.

## Implementation

We will achieve our goals through a series of progressive PRs
- [ ] use `pyproject.toml` to centralize and standardize project configuration.
- [ ] use `ruff `as the default linter and configure the rules
- [ ] use `ruff `as the default formatter and configure the line width
- [ ] specify the minimum Python version and configure it in `pyproject.toml` and `pyright` to standardize syntax and ensure compatibility
- [ ] configure linters in continuous integration to ensure that all PRs comply with specifications
- [ ] rethink the project structure, which may require further discussion 🤔
- [ ] unify naming conventions: modules should follow snake_case, while variables should adopt Qt’s smallCamelCase style.

**Comment by ChinaIceF** [comment]
I am very pleased with your concern for this project.  

In the past, due to the occupation of my time by my studies, I was unable to fully devote myself to the development of this project.  
But now, after the college entrance examination, I have more time to study and create, and constantly improve the project.  

I am very willing to work with you to complete these tasks.

**Comment by rainzee** [comment]
初步的项目标准化差不多了，接下来就是重新设计代码结构了？有兴趣讨论一下吗？

**Comment by ChinaIceF** [comment]
当然
我对这个库的定位是为用户提供模板（如gallery的界面）、控件（如按钮）和处理方法（如动画管理器）
所以我接下来打算将这三类东西分别放在``SiliconUI``文件夹下的三个模块文件夹内
同时我也打算将ui.py集成成一个类，用于快速在模板上构建界面

**Comment by rainzee** [comment]
有 QQ 吗，这种讨论还是在 IM 上高效一点，1004529162

**Comment by ChinaIceF** [comment]
好的

**Comment by Cassianvale** [comment]
Have you considered starting a discord group?

**Comment by ChinaIceF** [comment]
Yes, I think it will be convenient for both users and developers if start a group for discussing and exchanging ideas on IM platform. But I'm still hesitating about starting groups whether on QQ or on discord.
